tests/unit/features/partitions/test_lifecycle_manager.py

```diff
@@ -0,0 +1,138 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2023 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Unit tests for the lifecycle manager with the partitions feature."""
+
+from string import ascii_lowercase
+from typing import Any, Dict
+
+import pytest
+from hypothesis import HealthCheck, given, settings, strategies
+
+from craft_parts import errors
+from craft_parts.lifecycle_manager import LifecycleManager
+
+
+class TestPartitionsSupport:
+    """Verify LifecycleManager supports partitions."""
+
+    @pytest.fixture
+    def parts_data(self) -> Dict[str, Any]:
+        return {"parts": {"foo": {"plugin": "nil"}}}
+
+    @pytest.mark.parametrize("partitions", [["default"], ["default", "kernel"]])
```

Comment:
How about `["default", "default"]`

Reply:
(And as a follow-up, if duplicates aren't allowed, you probably want set your hypothesis strategy below to be unique and filter it to not contain `"default"`

Reply:
Also `["default", "kernel", "kernel"]` is a possibility.

Reply:
Great catch.  I added a check in the code and more tests.